crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -178,18 +168,27 @@ impl WorkerPool {
     }
 
     #[instrument(target = "miden-block-producer", skip_all, err, fields(batch_id))]
-    fn build_batch(&self, txs: Vec<ProvenTransaction>) -> Result<(), BuildBatchError> {
+    fn build_batch(
+        txs: Vec<AuthenticatedTransaction>,
+    ) -> Result<TransactionBatch, BuildBatchError> {
         let num_txs = txs.len();
 
         info!(target: COMPONENT, num_txs, "Building a transaction batch");
         debug!(target: COMPONENT, txs = %format_array(txs.iter().map(|tx| tx.id().to_hex())));
 
+        // TODO: This is a deep clone which can be avoided by change batch building to using
+        // refs or arcs.
+        let txs = txs
+            .iter()
+            .map(AuthenticatedTransaction::raw_proven_transaction)
+            .cloned()
+            .collect();
```

Comment:
I agree. I don't think there is a good reason why `TransactionBatch::new()` cannot take something like `IntoIterator<Item = AsRef<ProvenTransaction>` as input.

Reply:
Tracked in #531